src/vaadin-grid-styling-mixin.html

```diff
@@ -0,0 +1,73 @@
+<!--
+@license
+Copyright (c) 2018 Vaadin Ltd.
+This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+-->
+<script>
+  window.Vaadin = window.Vaadin || {};
+  window.Vaadin.Grid = window.Vaadin.Grid || {};
+
+  /**
+   * @polymerMixin
+   */
+  Vaadin.Grid.StylingMixin = superClass => class StylingMixin extends superClass {
+
+    static get properties() {
+      return {
+        /**
+         * A function that allows generating CSS class names for grid cells
+         * based on their row and column. The return value should be the generated
+         * class name as a string, or multiple class names separated by whitespace
+         * characters.
+         *
+         * Receives two arguments:
+         * - `rowData` The object with the properties related with
+         *   the rendered item, contains:
+         *   - `rowData.index` The index of the item.
+         *   - `rowData.item` The item.
+         *   - `rowData.expanded` Sublevel toggle state.
+         *   - `rowData.level` Level of the tree represented with a horizontal offset of the toggle button.
+         *   - `rowData.selected` Selected state.
+         * - `column` The `<vaadin-grid-column>` element.
+         */
+        cellClassNameGenerator: Function
+      };
+    }
+
+    static get observers() {
+      return [
+        '__cellClassNameGeneratorChanged(cellClassNameGenerator)'
+      ];
+    }
+
+    __cellClassNameGeneratorChanged(cellClassGenerator) {
+      this.generateClassNames();
+    }
+
+    /**
+     * Runs the `cellClassNameGenerator` for the visible cells.
+     * If the generator depends on varying conditions, you need to
+     * call this function manually in order to update the styles when
+     * the conditions change.
+     */
+    generateClassNames() {
+      Array.from(this.$.items.children).forEach(
```

Comment:
Note that iron-list items container may include hidden rows, for example after the items size is reduced from 2 to 1. So `this.$.items.children` should be filtered before processing.

Reply:
Done